### ISSUE TITLE
Because it has to be done manually, how does one setup SataPortMap and DiskIDxMap

### ISSUE THREAD

**Issue by Viking8** [fbelavenuto/arpl#115]
Hi,

I am new to ARPL. I am trying to get SataPortMap and DiskIDxMap setup, but I noticed that you didn't explain how to do this. Just that the cmdline area will help to so this.

If a model is chosen that uses the Device-tree system to define the HDs, there is no need to configure anything. **_In the case of models that do not use device-tree, the configurations must be done manually_** and for this there is an option in the "Cmdline" menu to display the SATA controllers, DUMMY ports and ports in use, to assist in the creation of the "SataPortMap", "DiskIdxMap" and "sata_remap" if necessary.

Could you reply with some help on how to set this up? Thank you.

**Comment by fbelavenuto** [comment]
Hi, please read this article:
https://gugucomputing.wordpress.com/2018/11/11/experiment-on-sata_args-in-grub-cfg/

**Comment by Viking8** [comment]
> Hi, please read this article: https://gugucomputing.wordpress.com/2018/11/11/experiment-on-sata_args-in-grub-cfg/

Hi Fabio, This article didn't really help me a whole lot. Could you provide a sample file(s) for me to study so I can get a better idea how the SATA info is structured. I'm sure if I see that along with this article it will help my brain make sense of it.

Thx!

**Comment by fbelavenuto** [comment]
Yes, please post a screenshot of the "Show SATA(s) # ports and drives" output.

**Comment by Viking8** [comment]
Hi Fabio, My setup is baremetal. I originally just had the built in SATA controller from my EVGA Z270 FTW K motherboard, but I added in an AsMedia 6 port PCI-e controller as well. I've attached screenshots of both. Thank you for your help.

![ARPL_02](https://user-images.githubusercontent.com/23368970/186323708-67bc7d55-18b7-4846-971f-1201f17605dd.png)

![ARPL_04](https://user-images.githubusercontent.com/23368970/186323748-8e8b4217-2f68-4339-9685-8f2e5ab4f999.png)

![ARPL_05](https://user-images.githubusercontent.com/23368970/186323760-a47f3c22-2cd3-4c3b-972e-e624f4f23c77.png)

**Comment by fbelavenuto** [comment]
In your case, you don't need to configure SataPortMap and DiskIdxMap.
Use the first ports so that the HDs are in sequence. Now if you don't want to use the onboard card you can disable it in the BIOS or if you can't there you can use the variables to configure it.
How do you intend to use these ports?
PS: I liked this card with 32 ports!! Do you have a link to buy this board?

**Comment by Viking8** [comment]
I'll try that and see if I can finally get past booting it. It always just sits there and then finally freezes. I never get an IP to set things up further. I thought it had to do with the drives not being detected as shown in the first pic I attached before.. As for that card, it only has 6 ports on board. I'm not sure why it's saying it has 32. Here's the US Amazon link for it.
 https://www.amazon.com/dp/B097NKJSDC

**Comment by Viking8** [comment]
Fabio, I was finally able to get ARPL to fully boot and install DSM 7.1 using the DS3622xs+. I do have a request though. Could you please send me a sample file of what it would look like when SataPortMap and DiskIdxMap need to be set? Is it in the user config file or a different file all together? It just isn't clear to me how to do this if I would need to for a different build. And exactly how would I know if I need to set those things up or not? Thanks.

**Comment by fbelavenuto** [comment]
32 ports that appear here:
![image](https://user-images.githubusercontent.com/11461290/186646627-2d3986ff-2128-42dc-b1ae-db9f3c8aa651.png)
Now I understand your question, here's how to do it:
![image](https://user-images.githubusercontent.com/11461290/184919976-60e8704f-2cef-4e79-b73e-f3b210e662eb.png)
![image](https://user-images.githubusercontent.com/11461290/184920048-cf71343f-cb41-40cd-8156-4f82858fca73.png)
![image](https://user-images.githubusercontent.com/11461290/184920130-b110bda0-30a8-4291-a048-fef290befd95.png)
![image](https://user-images.githubusercontent.com/11461290/184920204-b5be5fa2-4a3a-4ad5-a722-5f5a65e8984e.png)
Same to DiskIdxMap
All the configuration is in a single file called user-config.yml on the first partition, which is accessible on a windows machine. There is a way to edit manually through the menu itself, but it is not necessary as it is possible to do everything through the menu. One sample for curiosity:
![image](https://user-images.githubusercontent.com/11461290/186648129-84e7f504-c83c-4dbf-8b6f-bd27f04a9353.png)

I have difficulty with the English language, so I didn't quite understand your question, sorry for that.
I believe I have now managed to clear your doubts?

**Comment by sam5558** [comment]
Hi,
I'm having difficulties to guess which DiskIdxMap & SataPortMap to apply.
Can someone help me please ?

![Screenshot_20220825_123742](https://user-images.githubusercontent.com/26897744/186654370-4f75ac4c-12e9-4556-b21e-245ad6fe9173.png)
![Screenshot_20220825_123803](https://user-images.githubusercontent.com/26897744/186654451-f50d7ac7-4ce1-4cf4-8bf1-5db90b65514b.png)

**Comment by fbelavenuto** [comment]
In your case it is not necessary to define the variables, please delete them (DiskIdxMap and SataPortMap).
This error could be related to something else.

**Comment by sam5558** [comment]
Thank you for your reply @fbelavenuto (and btw thanks also for providing the community with this awesome project) ,
I've created a new VM from scratch and avoided to create any custom variable as you said.
The build gave me some errrors in red, but just after that it gave me the possibility of booting the image, which i did.
![Screenshot_20220825_153738](https://user-images.githubusercontent.com/26897744/186694534-b3f9969d-0344-4438-8103-77c1a6afebe9.png)

No issue observed on the boot, however, the ip (192.168.240.7) is no longer accessible, and just before the boot the ip was working... kinda strange no ?

So right now my issue is that the IP isn't pinging. I suspected a driver issue but it doesn't make sense as i were able to configure the whole thing through network via http://192.168.240.7:7681/

PS : I've added virtio.addon because i'm virtualizing on unRAID with qEMU & virtio + sata drives.

Do you have an idea ?

**Comment by fbelavenuto** [comment]
Loades uses a different, more up-to-date linux kernel. DSM uses different Linux kernel. The problem must be in the driver compiled for the DSM, that's why it only works in the loader.
Please show us any errors that exist when creating the loader

**Comment by sam5558** [comment]
Hi again @fbelavenuto ,

Here is a screenshot of what happens when i build the loader : 

![Screenshot_20220825_171931](https://user-images.githubusercontent.com/26897744/186718578-2f06fa79-6348-412f-ae0d-6bd2afc5c718.png)

![Screenshot_20220825_172102](https://user-images.githubusercontent.com/26897744/186718542-77cbe22b-30ee-47dd-a9f5-5adf0b572214.png)

**Comment by fbelavenuto** [comment]
Ok, put more RAM to your virtual machine, minimal 4GB

**Comment by sam5558** [comment]
Thanks @fbelavenuto , adding the RAM to 4GB makes the build succeed !

![Screenshot_20220825_173301](https://user-images.githubusercontent.com/26897744/186720893-94ee3c9d-221a-461d-aa6f-83402159c174.png)

I hate myself for bothering you again but now, within the Synology process, DSM can't install : 
![Screenshot_20220825_173307](https://user-images.githubusercontent.com/26897744/186720978-5fb69ae4-323b-47d9-b339-7136a3706cb6.png)

![Screenshot_20220825_173328](https://user-images.githubusercontent.com/26897744/186721024-c4286de9-a104-4a0a-ac67-8f6f5178ffcd.png)

Do i have to find a way and format that disk myself ?

**Comment by fbelavenuto** [comment]
Try again. If trouble yet, choose the option "Automatically download and install..."

**Comment by sam5558** [comment]
I did try both of the options, but somehow i get the same error.

The thing is the vdisk i'm using is new, just created it, so i don't understand where is the issue.

If i boot with gParted (iso) and try to format the target virtual drive, is it gonna help ?

**Comment by fbelavenuto** [comment]
Try put 2 or more data HDDs

**Comment by sam5558** [comment]
I've added 2 other HDDs, doesn't seem to help : 
![Screenshot_20220825_175527](https://user-images.githubusercontent.com/26897744/186725094-47d1bcf3-5681-4eb0-afd4-5d503659832f.png)

![Screenshot_20220825_175538](https://user-images.githubusercontent.com/26897744/186725131-e178b2d2-08a9-45d3-9ea6-831ce07ebffa.png)

Let's call it a day. I'll try again maybe tomorrow.
Anyway thank you so much @fbelavenuto for your help and bearing with me.

**Comment by fbelavenuto** [comment]
How is a VM, add a serial port, save the log and upload here.

Em qui., 25 de ago. de 2022 às 13:58, sam5558 ***@***.***>
escreveu:

**Comment by sam5558** [comment]
Hi @fbelavenuto ,
The reason why it kept on failing on the installation part was because i previously created vdisks below 10G. 
I created a vDisk of 45G, now everything is working.

Thank you so much for the project & for helping me out ! :+1:

**Comment by fbelavenuto** [comment]
Ok, good to know!! Closing issue

**Comment by Viking8** [comment]
Thanks for the help Fabio. I do have another question. How do you know what values to give to SataPortMap and DiskIdxMap?
In the example you gave you had "66" and "0600". 

![image](https://user-images.githubusercontent.com/23368970/187021221-12740cf7-58e5-49b6-bfd6-4fdc1a3af86e.png)

**Comment by fbelavenuto** [comment]
SataPortMap configures how many SATA ports each adapter will have. Each digit tells the number of ports, it cannot be zero and the maximum is 9. In this case, the first adapter will have 6 ports and the second 6 ports. If it was "SataPortMap: 125" it means that the first adapter will have 1 port only, the second only 2 ports and the third 5 ports.
DiskIdxMap is used to say when the disk count starts, there are 2 digits for each adapter. In this case "DiskIdxMap: 0600" means that the first adapter will start as the sixth disk (A=0, B=1..G=6), ie the first disk of the first adapter will be named "sdg". The second adapter will start counting from zero, that is, the first disk of the second adapter will be named "sda".
In short, this example configuration basically reverses the placeholders.
